core/iwasm/libraries/lib-socket/src/wasi/wasi_socket_ext.c

```diff
@@ -78,8 +78,26 @@ wasi_addr_to_sockaddr(const __wasi_addr_t *wasi_addr,
             break;
         }
         case IPv6:
-            // TODO: IPV6
-            return __WASI_ERRNO_AFNOSUPPORT;
+        {
+            struct sockaddr_in6 sock_addr_in6 = { 0 };
+            uint16_t *addr_buf = (uint16_t *)sock_addr_in6.sin6_addr.s6_addr;
+
+            addr_buf[0] = htons(wasi_addr->addr.ip6.addr.n0);
+            addr_buf[1] = htons(wasi_addr->addr.ip6.addr.n1);
+            addr_buf[2] = htons(wasi_addr->addr.ip6.addr.n2);
+            addr_buf[3] = htons(wasi_addr->addr.ip6.addr.n3);
+            addr_buf[4] = htons(wasi_addr->addr.ip6.addr.h0);
+            addr_buf[5] = htons(wasi_addr->addr.ip6.addr.h1);
+            addr_buf[6] = htons(wasi_addr->addr.ip6.addr.h2);
+            addr_buf[7] = htons(wasi_addr->addr.ip6.addr.h3);
```

Comment:
Had better add APIs to do the conversion of addr_buf and addr after PR #1411 is merged